routers/users-router.js

```diff
@@ -0,0 +1,108 @@
+express = require('express');
+
+const db = require('../data/helpers/userDb.js');
+
+const router = express.Router();
+
+//Custom middleware
+const bigName = (req, res, next) => {
+    name = req.body.name.toUpperCase();
+    next();
+}
+
+//Get
+router.get('/', (req, res) => {
+    db
+        .get()
+        .then(users => {
+            res.status(200).json(users)
+        })
+        .catch(err => {
+            res.status(500).json({ error: 'The users information could not be retrieved.' })
+        });
+});
+
+//Get(user)
+router.get('/:id', (req, res) => {
+    const id = req.params.id;
+    db
+        .getById(id)
+        .then(user => {
+            if(!user) {
+                return res.status(404).json({ message: 'The user with the specified ID does not exist.' })
+            }
+            res.status(202).json(user)
+        })
+        .catch(err => {
+            res.status(500).json({ error: 'The user could not be retrieved.'})
+        });
+});
+
+//Get(user-posts)
+router.get('/posts/:userId', (req, res) => {
```

Comment:
I would recommend this instead
```suggestion
router.get('/:id/posts, (req, res) => {
```